sound/soc/intel/common/soc-acpi-intel-ptl-match.c

```diff
@@ -134,6 +152,12 @@ struct snd_soc_acpi_mach snd_soc_acpi_intel_ptl_sdw_machines[] = {
 		.drv_name = "sof_sdw",
 		.sof_tplg_filename = "sof-ptl-rt722.tplg",
 	},
+	{
+		.link_mask = BIT(1),
+		.links = ptl_rt722_l1,
+		.drv_name = "sof_sdw",
+		.sof_tplg_filename = "sof-ptl-rt722.tplg",
```

Comment:
the problem I have is that this configuration requires a a 4ch DMIC capture stream in the topology, but the name of the topology is not quite clear, whereas the topology on line 153 doesn't need the DMIC capture.

And btw why do we have TWO entries for the same RVP? Which one is correct here?

Reply:
@plbossart how are you saying that the config requires a 4ch DMIC from this entry?

Reply:
@ranj063 : I think i understood the question from Pierre.
so, if we are using PDM DMIC then the following change is required in topology and so the topology name change to sof-ptl-rt722-pdmdmic.tplg is what he is expecting which can have the below config incorporated. @plbossart, hope my undertsanding is right

-"cavs-sdw\;sof-ptl-rt722\;PLATFORM=ptl,SDW_DMIC=1,NUM_SDW_AMP_LINKS=1,\
+"cavs-sdw\;sof-ptl-rt722\;PLATFORM=ptl,SDW_DMIC=1,NUM_SDW_AMP_LINKS=1,NUM_DMICS=2,\
+PDM1_MIC_A_ENABLE=0,PDM1_MIC_B_ENABLE=0,DMIC0_ID=5,DMIC1_ID=6,HDMI1_ID=7,HDMI2_ID=8,HDMI3_ID=9,\
 SDW_AMP_FEEDBACK=false,SDW_SPK_STREAM=Playback-SmartAmp,SDW_DMIC_STREAM=Capture-SmartMic,\
-SDW_JACK_OUT_STREAM=Playback-SimpleJack,SDW_JACK_IN_STREAM=Capture-SimpleJack"
+SDW_JACK_OUT_STREAM=Playback-SimpleJack,SDW_JACK_IN_STREAM=Capture-SimpleJack,\
+PREPROCESS_PLUGINS=nhlt,NHLT_BIN=nhlt-sof-ptl-rt722.bin"

Reply:
The rule is that you can enable a dailink in the machine driver, and if the topology does not refer to this dailink that dailink will be ignored. So if you do want the DMIC functionality to be available, then you do need a matching topology change indeed.

But the problem here is that your change would add the DMIC path to ALL platforms that rely on RT722. I am not sure it's a wise thing to do, it's quite possible that the BIOS menu does not enable such DMICs, or that there are pin-mux conflicts. In both cases the driver would end-up exposing a PCM path that's not functional.

So ideally we would want this machine driver quirk and topology change only used for Chrome-specific validation. I don't know though how we can identify this configuration, we'd need a more focused filter rather than the blanket filter on the DMI platform name.

Reply:
@RDharageswari, I guess we need two topologies, right?
`sof-ptl-rt722.tplg` for rt7222 on link0 or link1
`sof-ptl-rt722-4ch.tplg` for rt722 (on link1) and 4 channel PCH-DMIC

With Chrome you need the kernel parameter, with EFI the kernel will detect the DMICs and append the -4ch to the topology file name.

Reply:
> @RDharageswari, I guess we need two topologies, right? `sof-ptl-rt722.tplg` for rt722 on link0 or link1 `sof-ptl-rt722-4ch.tplg` for rt722 (on link1) and 4 channel PCH-DMIC
> 
> With Chrome you need the kernel parameter, with EFI the kernel will detect the DMICs and append the -4ch to the topology file name.

The -4ch will be append whenever the PCH-DMIC is used. We need the kernel parameter for Chrome. With EFI the kernel will detect the DMICs and no kernel parameter is needed.

Reply:
https://github.com/thesofproject/sof/pull/9493/commits/78b74f7da00e44d843c3798546d7c552a2ed7c83
i have submitted here the sof-ptl-rt722-4ch.tplg. 
And yes -4ch will be added dynamically on looking at the module params